code-generator/core/src/main/java/com/tigrisdata/tools/validation/PrimaryKeysRule.java

```diff
@@ -0,0 +1,37 @@
+package com.tigrisdata.tools.validation;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class PrimaryKeysRule implements ValidationRule {
+
+  private static final String PRIMARY_KEYS = "primary_key";
+  private static final String PROPERTIES = "properties";
+
+  @Override
+  public void validate(JsonObject beforeSchema, JsonObject afterSchema) throws ValidationException {
+    JsonArray beforePrimaryKeys = beforeSchema.get(PRIMARY_KEYS).getAsJsonArray();
```

Comment:
what if beforeSchema is nil, is there any check for that somewhere?

Reply:
yes that is checked at higher level application in maven plugin. code-generator/maven-plugin/src/main/java/com/tigrisdata/maven/SourceCodeGeneratorMojo.java

Reply:
sg